report_qweb_decimal_place/readme/DESCRIPTION.rst

```diff
@@ -0,0 +1,28 @@
+This module intends to provide the base function for currencies to adjust the number of decimal places
+for the unit price in QWeb reports.
+Installing this module alone does not affect the presentation of existing QWeb reports.
+Individual adjustments need to be done in separate modules in a manner similar to the following:
+
+.. code-block:: xml
+
+    <template
+        id="report_saleorder_document_inherit"
+        inherit_id="sale.report_saleorder_document"
+    >
+        <xpath expr="//td[@name='td_priceunit']/span" position="replace">
+                <t t-set="currency" t-value="doc.currency_id" />
+                <t t-set="price_unit" t-value="line.price_unit" />
+                <t t-call="report_qweb_decimal_place.price_unit_value_format" />
+        </xpath>
+    </template>
+
+Background:
+~~~~~~~~~~~
+
+Odoo default reports display price unit with the decimal accuracy of product price configuration.
+However, globally applying the decimal accuracy setting is sometimes not appropriate under multi-currency settings
+where how unit prices should be presented differs depending on the currency.
+
+For example, unit prices in JPY usually do not have decimals (with some exceptions depending on the industry),
+while those in USD may require up to 2-4 decimals.  If we configure the decimal accuracy based on USD, the unit price
+presentation on PDF reports for JPY transactions may appear  a bit unconventional.
```

Comment:
Hm, but if you look at for example the display of subtotal in Odoo 15's [sale report](https://github.com/OCA/OCB/blob/15.0/addons/sale/report/sale_report_templates.xml#L134), it uses `                                    <span t-esc="current_subtotal" t-options='{"widget": "monetary", "display_currency": doc.pricelist_id.currency_id}'/>`.

Under the hood, the [monetary widget](https://github.com/OCA/OCB/blob/15.0/odoo/addons/base/models/ir_qweb_fields.py#L433) uses the currency to determine the amount of decimal places to display.

So, in the above example, why wouldn't you go for:

```
        <xpath expr="//td[@name='td_priceunit']/span" position="replace">
                <t t-esc="line.price_unit" t-options='{"widget": "monetary", "display_currency": doc.currency_id}' />
        </xpath>
```

Reply:
Thanks for the review. But the purpose of this module is not only for using same decimal place with the currency. Also for dynamic decimal place for price unit in pdf reports.